FirebaseCore/Sources/FIRBundleUtil.m

```diff
@@ -49,15 +49,18 @@ + (NSArray *)relevantURLSchemes {
 
 + (BOOL)hasBundleIdentifierPrefix:(NSString *)bundleIdentifier inBundles:(NSArray *)bundles {
   for (NSBundle *bundle in bundles) {
-    // This allows app extensions that have the app's bundle as their prefix to pass this test.
-    NSString *applicationBundleIdentifier =
-        [GULAppEnvironmentUtil isAppExtension]
-            ? [self bundleIdentifierByRemovingLastPartFrom:bundle.bundleIdentifier]
-            : bundle.bundleIdentifier;
-
-    if ([applicationBundleIdentifier isEqualToString:bundleIdentifier]) {
+    if ([bundle.bundleIdentifier isEqualToString:bundleIdentifier]) {
       return YES;
     }
+
+    if ([GULAppEnvironmentUtil isAppExtension]) {
+      // This allows app extensions that have the app's bundle as their prefix to pass this test.
```

Comment:
Nit: I think we should describe why we're doing this, not what the code is doing. Something like:

```
A developer could be using the same `FIROptions` for both their app and extension. Since extensions have a suffix added to the bundleID, we consider a matching prefix as valid.
```